src/input/src/joystick.rs

```diff
@@ -0,0 +1,68 @@
+
+//! Back-end agnostic joystick events.
+
+use std::any::Any;
+
+use { GenericEvent, JOYSTICK_AXIS };
+
+/// The position of a joystick axis changed.
+pub trait JoystickAxisEvent: Sized {
+    /// Creates a joystick axis event.
+    fn from_wav(which: i32, axis: u8, value: f64, old_event: &Self) -> Option<Self>;
```

Comment:
`from_which_axis_value` is more readable than using an abbreviation `from_wav`.

Perhaps we should add a `JoystickArgs` struct?